### ISSUE TITLE
Reply UI bugs

### ISSUE THREAD

**Issue by guidezpl** [flutter/gallery#567]
- [ ] Removing an email from inbox through the action item, does not animate: When an email is deleted through the action bar located in the MailPreviewCard on larger screens, and the BottomAppBarActionItems on smaller screens, there is no animation to show the email being dismissed from the inbox. The email instantly dissapears.
- [ ] Bottom drawer destinations not visible on devices with small screens: The bottom drawer on the reply material study is not a scrollable widget. So, if a device's screen is too small to fit the content of the bottom drawers destinations, then that device won't be able to navigate to said destinations.
- [ ] Contextual bottom app bar icons don't fade in from the center: The icons in the contextual bottom app bar look like they fade in from the right of the screen, rather than coming in from the center.
- [ ] Bottom drawer animations are not consistent: The bottom drawer closes too fast, it is currently set at a duration of 300ms, but it closes way faster. The Floating action button should also enter/exit smoothly with the bottom drawer. Floating action button animation should use standardEasing.
- [ ] Mobile FAB uses SizedBox instead of FAB: Because of an a11y issue with the animations package, we use a SizedBox instead of a fab. A SizedBox does not retain the visual states of a FAB so this is a workaround until this issue is fixed in the animations package.
- [ ] Use onGenerateRoute for navigating to SearchPage: In order to navigate to the SearchPage we use a PageTransitionSwitcher and the onSearchPage property in our EmailStore, to switch between the SearchPage and the HomePage. Originally this was implemented through onGenerateRoute, but because of a bug a workaround was put in place. [PR](https://github.com/Renzo-Olivares/gallery/pull/47).
- [ ]  Fade through transition between navigation rail destinations broken: The fade through transition that occurs when tapping on different navigation rail destinations is not working. Tapping different destinations does not yield a transition.
